src/components/Select/SelectT.tsx

```diff
@@ -120,34 +141,24 @@ export default function SelectT<T>(props: SelectTProps<T>): JSX.Element {
     ) {
       setOpenedOptions(false);
     }
-  };
+  }, []);
 
-  const handleResize = () => {
+  const handleResize = useCallback(() => {
     setOpenedOptions(false);
-  };
-
-  const hasOptions = useMemo<boolean>(() => options !== undefined &&  options.length > 0, [options]);
+  }, []);
 
   useEffect(() => {
-    if (openedOptions && hasOptions) {
-      scrollToSelectedElement();
-      if (fixedMenu && inputRef.current && dropdownRef.current) {
-        dropdownRef.current.style.width = `${inputRef.current.offsetWidth}px`;
-        const bbox = inputRef.current.getBoundingClientRect();
-        dropdownRef.current.style.top = `${bbox.top + bbox.height}px`;
-        const dropdownBottom = dropdownRef.current.clientHeight + bbox.top + bbox.height;
-        const windowHeightThreshold = window.innerHeight - bbox.height;
-        if (dropdownBottom > windowHeightThreshold) {
-          dropdownRef.current.style.maxHeight = `${
-            dropdownRef.current.clientHeight - (dropdownBottom - windowHeightThreshold)
-          }px`;
-        }
-        setFixedMenuPositioned(true);
-      }
-    } else if (fixedMenu) {
-      setFixedMenuPositioned(false);
-    }
-  }, [fixedMenu, openedOptions, hasOptions]);
+    window.addEventListener('click', handleClick);
```

Comment:
This was moved down to enable declaration of useCallback dependencies, absence of which was causing behavior issues.